contracts/StrategyVersionRegistry.vy

```diff
@@ -0,0 +1,85 @@
+# @version 0.2.12
+governance: public(address)
+pendingGovernance: public(address)
+
+event NewGovernance:
+    governance: address
+
+event NewPendingGovernance:
+    governance: address
+
+event StrategyRegistered:
+    strategy: address
+    name: String[64]
+    apiVersion: String[8]
+
+event Cloned:
+    clone: address
+
+interface Strategy:
+    def name() -> String[64]: view
+    def apiVersion() -> String[8]: view
+    def initialize(params: Bytes[128]): nonpayable
+    def isOriginal() -> bool: nonpayable
+
+strategyVersions: public(HashMap[String[73], address])
+SEPARATOR: constant(String[1]) = "@"
+
+@external
+def __init__():
+    self.governance = msg.sender
+
+@external
+def setGovernance(governance: address):
+    """
+    @notice Starts the 1st phase of the governance transfer.
+    @dev Throws if the caller is not current governance.
+    @param governance The next governance address
+    """
+    assert msg.sender == self.governance  # dev: unauthorized
+    log NewPendingGovernance(msg.sender)
+    self.pendingGovernance = governance
+
+
+@external
+def acceptGovernance():
+    """
+    @notice Completes the 2nd phase of the governance transfer.
+    @dev
+        Throws if the caller is not the pending caller.
+        Emits a `NewGovernance` event.
+    """
+    assert msg.sender == self.pendingGovernance  # dev: unauthorized
+    self.governance = msg.sender
+    log NewGovernance(msg.sender)
+
+@external
+def addNewRelease(strategy :address, name: String[64] = ""):
+    """
+    @notice add a new version of a strategy
+    """
+    assert msg.sender == self.governance  # dev: unauthorized
+    assert Strategy(strategy).isOriginal() # dev: not original 
+    strategyName: String[64] = name 
+    if name == "":
+        strategyName = Strategy(strategy).name()
+        assert strategyName != ""
+    apiVersion: String[8] = Strategy(strategy).apiVersion()
+    key: String[73] = concat(strategyName, SEPARATOR, apiVersion)
```

Comment:
Might be better to abstract this to it's own internal function
```python
@pure
@internal
def _computeKey(name: String[64], version: String[8]) -> String[73]:
    return concat(name, "@", version)  # NOTE: No need for constant here
```

And then just use it via:
```suggestion
    key: String[73] = self._computeKey(strategyName, apiVersion)
```